File: projects/GPL-FH-Java/features/DirectedWithNeighbors/GPL/Vertex.java
package GPL;

import java.util.Iterator;
import java.util.LinkedList;
// of Graph
   
  // *************************************************************************   
   
public class Vertex {
    public LinkedList adjacentNeighbors;
    public String name;
   
    public Vertex() {
        VertexConstructor();
    }
    public String getName( ) 
    { 
        return name; 
    }

    public void VertexConstructor() {
        name      = null;
        adjacentNeighbors = new LinkedList();
    }

    public  Vertex assignName( String name ) {
        this.name = name;
        return ( Vertex ) this;
    }
   
    public void addEdge( Neighbor n ) {
        adjacentNeighbors.add( n );
    }


    public VertexIter getNeighbors( ) 
    {
        return new VertexIter( ) 
        {
            private Iterator iter = adjacentNeighbors.iterator( );
            public Vertex next( ) 
            { 
               return ( ( Neighbor )iter.next( ) ).neighbor; 
            }

            public boolean hasNext( ) 
            {
               return iter.hasNext( ); 
            }
        };
    }

    public void adjustAdorns( Neighbor sourceNeighbor )
      {}
      
    public void display() 
    {
        System.out.print( "Node " + getName( ) + " connected to: " );

        for(VertexIter vxiter = getNeighbors( ); vxiter.hasNext( ); )
         {
            Vertex v = vxiter.next( );
            System.out.print( v.getName( ) + ", " );
        }
        System.out.println( );
    }
}

File: projects/GPL-FH-CSharp/features/DirectedWithNeighbors/Vertex.cs
//layer Directed;

//import java.util.Iterator;
//import java.util.LinkedList;
// of Graph
   
  // *************************************************************************   
   using System.Collections;
   using System.Collections.Generic;
public class Vertex {
    public List<Vertex> adjacentNeighbors;
    public string name;
   
    public Vertex() {
        VertexConstructor();
    }
    public string GetName( ) 
    { 
        return name; 
    }

    public virtual void VertexConstructor() {
        name      = null;
        adjacentNeighbors = new List<Vertex>();
    }

    public  virtual Vertex assignName( string name ) {
        this.name = name;
        return ( Vertex ) this;
    }
   
    public virtual void AddEdge( Neighbor n ) {
        adjacentNeighbors.Add(n);
    }


    public virtual Iterator<Vertex> GetNeighbors( ) 
    {
        return new Iterator<Vertex>(adjacentNeighbors.GetEnumerator(), TransformToNeighbor);
    } 
    public static Vertex TransformToNeighbor (Vertex val) {
    	return ((Neighbor)val.Current).neighbor;
    }

    public virtual void adjustAdorns( Neighbor sourceNeighbor )
      {}
      
    public virtual void display() 
    {
        System.Console.Out.Write( "Node " + GetName( ) + " connected to: " );

        for(Vertex<Iterator> vxiter = GetNeighbors( ); vxiter.hasNext( ); )
         {
            Vertex v = vxiter.next( );
            System.Console.Out.Write( v.GetName( ) + ", " );
        }
        System.Console.Out.WriteLine( );
    }
}
